packages/console/src/components/CodeEditor/index.tsx

```diff
@@ -12,11 +12,44 @@ type Props = {
   isReadonly?: boolean;
   value?: string;
   onChange?: (value: string) => void;
+  tabSize?: number;
 };
 
-const CodeEditor = ({ className, language, isReadonly = false, value = '', onChange }: Props) => {
+const CodeEditor = ({
+  className,
+  language,
+  isReadonly = false,
+  value = '',
+  onChange,
+  tabSize = 4,
+}: Props) => {
+  const [text, setText] = useState<{ caret: number; target?: HTMLTextAreaElement }>({
+    caret: -1,
+    target: undefined,
+  });
```

Comment:
First of all, the state `text` seems inaccurate, as the state itself seems to have nothing to with `text`.
Moreover, I don't think we need to set target to state. We should use `useRef` instead to keep the reference of the textarea.

Threfore, IMO the only thing that should be put in the state is the position of the caret. Consider changing it to:
```ts
const [caretPosition, setCaretPosition] = useState(-1);
```

Reply:
I agree I have applied your sugesstions.